packages/core/lib/testing/testrunner.js

```diff
@@ -49,231 +50,164 @@ function TestRunner(options = {}) {
-  var self = this;
+TestRunner.prototype.resetState = async function() {
   if (this.can_snapshot) {
-    this.revert(this.initial_snapshot, function(err) {
-      if (err) return callback(err);
-      self.snapshot(function(err, snapshot) {
-        if (err) return callback(err);
-        self.initial_snapshot = snapshot;
-        callback();
-      });
-    });
+    await this.revert(this.initial_snapshot);
+    this.initial_snapshot = await this.snapshot();
   } else {
-    this.deploy(callback);
+    await this.deploy();
   }
 };
 
-TestRunner.prototype.startTest = function(mocha, callback) {
-  this.interfaceAdapter
-    .getBlockNumber()
-    .then(result => {
-      const one = web3Utils.toBN(1);
-      const resultBN = web3Utils.toBN(result);
-
-      // Add one in base 10
-      this.currentTestStartBlock = resultBN.add(one);
+TestRunner.prototype.startTest = async function() {
+  let blockNumber = await this.web3.eth.getBlockNumber();
+  let one = web3Utils.toBN(1);
+  blockNumber = web3Utils.toBN(blockNumber);
 
-      callback();
-    })
-    .catch(callback);
+  // Add one in base 10
+  this.currentTestStartBlock = blockNumber.add(one);
 };
 
-TestRunner.prototype.endTest = function(mocha, callback) {
-  var self = this;
-
+TestRunner.prototype.endTest = async function(mocha) {
   // Skip logging if test passes and `show-events` option is not true
-  if (mocha.currentTest.state !== "failed" && !self.config["show-events"]) {
-    return callback();
+  if (mocha.currentTest.state !== "failed" && !this.config["show-events"]) {
+    return;
   }
 
-  // There's no API for eth_getLogs?
-  this.rpc(
-    "eth_getLogs",
-    [
-      {
-        fromBlock: "0x" + this.currentTestStartBlock.toString(16)
-      }
-    ],
-    function(err, result) {
-      if (err) return callback(err);
-
-      var logs = result.result;
-
-      if (logs.length === 0) {
-        self.logger.log("    > No events were emitted");
-        return callback();
-      }
-
-      self.logger.log("\n    Events emitted during test:");
-      self.logger.log("    ---------------------------");
-      self.logger.log("");
-
-      logs.forEach(function(log) {
-        var event = self.known_events[log.topics[0]];
+  function printEvent(decoding) {
+    const anonymousPrefix = decoding.kind === "anonymous" ? "<anonymous> " : "";
+    const className = decoding.class.typeName;
+    const eventName = decoding.abi.name;
+    const fullEventName = anonymousPrefix + `${className}.${eventName}`;
+    const eventArgs = decoding.arguments
+      .map(({ name, indexed, value }) => {
+        let namePrefix = name ? `${name}: ` : "";
+        let indexedPrefix = indexed ? "<indexed> " : "";
+        let displayValue = util.inspect(
+          new Codec.Format.Utils.Inspect.ResultInspector(value),
+          {
+            depth: null,
+            colors: true,
+            maxArrayLength: null,
+            breakLength: Infinity,
```

Comment:
Can we make breakLength be 80?
```suggestion
            breakLength: 80,
```

(or whatever, to get the lines to be a bit shorter... I guess this might require finagling to get the indent to work right)

Reply:
Sure!  I was just matching the old behavior which was printing the whole thing on one line.  As you say, though, splitting over multiple lines will require some indent finagling.

Reply:
OK, I did this, but in order for the result to make sense I also had to put each argument on a separate line.  And so I also ended up putting a blank line between events like you suggested elsewhere.  Also, I decided to reintroduce the type marker, because why not.

I am hoping the result is good?  I encourage you to try it out and see what you think.